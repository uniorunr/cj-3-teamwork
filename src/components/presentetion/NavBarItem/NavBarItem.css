li {
  list-style: none;
  height: 100%;
  width: 50%;
}

li a {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  text-decoration: none;
  transition: 0.3s;
  text-transform: uppercase;
  color: #000;
  font-weight: bold;
  font-family: 'Open Sans';
}

li a:hover {
  background: rgba(255, 255, 255, 0.1);
  box-shadow: 0 0 10px 0 #000;
}
.active {
  width: 75%;
}

.not-active {
  width: 25%;
  background-color: #000;
}

.not-active a {
  color: #fff;
}
