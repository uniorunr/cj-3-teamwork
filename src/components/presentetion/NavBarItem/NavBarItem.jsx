import React from 'react';
import PropTypes from 'prop-types';
import './NavBarItem.css';

const NavBarItem = ({
  id, active, name, onClick,
}) => (
  <li className={active}>
    <a href={`#${id}`} id={id} onClick={onClick}>
      {name}
    </a>
  </li>
);

NavBarItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  active: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default NavBarItem;
