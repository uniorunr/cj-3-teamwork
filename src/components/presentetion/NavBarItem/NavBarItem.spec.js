import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import NavBarItem from './NavBarItem';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <NavBarItem id="home" hash="/" onClick={() => {}} name="Home" />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
