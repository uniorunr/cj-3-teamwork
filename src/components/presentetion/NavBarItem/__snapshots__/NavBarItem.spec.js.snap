// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="active"
>
  <a
    href="#home"
    id="home"
    onClick={[Function]}
  >
    Home
  </a>
</li>
`;
